### ISSUE TITLE
Unarchive fails with cryptic message

### ISSUE THREAD

**Issue by leonardolang** [ansible/ansible-modules-core#4660]
##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

unarchive
##### ANSIBLE VERSION

```
ansible 2.2.0
  config file = /root/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
- **NOTE: Using ansible from git, commit c8b8cda (lastest version) - but issue happening at least since commit d54f527**
##### CONFIGURATION

```
[defaults]
remote_user = root

[ssh_connection]
ssh_args = -o ForwardAgent=yes
scp_if_ssh = True
```

No ANSIBLE_\* variables set.
##### OS / ENVIRONMENT
- master = Arch Linux
- remote = CentOS 7
##### SUMMARY

Unarchive fails with an odd error message.

Filename shown is completely different from filename defined on task.

Using relative or absolute paths makes no difference in the result.
##### STEPS TO REPRODUCE
1. Create a playbook with this task:

```
- name: Uncompress tarball
  unarchive:
    src=blah.tar.gz
    dest={{ my_root_dir }}
    copy=yes
    creates=yes
    owner={{ my_user }}
    group={{ my_group }}
```
1. Place source file in "files" directory of role.
2. Run the playbook.
##### EXPECTED RESULTS

Unarchive should work.
##### ACTUAL RESULTS

Output with "ansible-playbook -vvv":

```
fatal: [10.1.2.3]: FAILED! => {
    "changed": false, 
    "failed": true, 
    "invocation": {
        "module_args": {
            "backup": null, 
            "content": null, 
            "copy": true, 
            "creates": "yes", 
            "delimiter": null, 
            "dest": "/opt/blah/public/", 
            "directory_mode": null, 
            "exclude": [], 
            "extra_opts": [], 
            "follow": false, 
            "force": null, 
            "group": "blah", 
            "keep_newer": false, 
            "list_files": false, 
            "mode": null, 
            "original_basename": "blah.tar.gz", 
            "owner": "blah", 
            "regexp": null, 
            "remote_src": false, 
            "selevel": null, 
            "serole": null, 
            "setype": null, 
            "seuser": null, 
            "src": "/root/.ansible/tmp/ansible-tmp-1472849250.29-49307224828763/source", 
            "unsafe_writes": null, 
            "validate_certs": true
        }
    }, 
    "msg": "Failed to find handler for \"/root/.ansible/tmp/ansible-tmp-1472849250.29-49307224828763/source\". Make sure the required command to extract the file is installed.", 
    "stat": {
        "exists": false
    }
}
```

**Comment by ansibot** [comment]
@dagwieers, @pileofrogs, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by dagwieers** [comment]
It is in fact not a cryptic message (at least if you look what the code does). The file type is not supported. It claims to be a **tar.gz** file, but either it is not, or you lack the commands to support the file. (gtar, bzip2 ?)

Or this is a bug, and we need more information (either to reproduce it ourselves, or to understand what is going on). needs_info

**Comment by leonardolang** [comment]
@dagwieers About your questions:
- `tar -zxvf` successfully extracts the archive. More info about the archive/system:

```
# file roles/blah/files/blah.tar.gz 
roles/blah/files/blah.tar.gz: gzip compressed data, last modified: Thu Aug 25 21:03:38 2016, from Unix
```

```
# gzip
gzip: compressed data not written to a terminal. Use -f to force compression.
```

```
# tar
tar: You must specify one of the '-Acdtrux', '--delete' or '--test-label' options
Try 'tar --help' or 'tar --usage' for more information.
```
- I didn't specify "/root/.ansible/tmp/ansible-tmp-1472849250.29-49307224828763/source" as the source archive, I specified "blah.tar.gz".
  
  So, at the very least, some internal mechanism-specific information is leaking to the outside - thus the term "cryptic" - without any reference to the original filename.

Reiterating: I'm using ansible from git (see commit ids on my previous message).

**Comment by dagwieers** [comment]
Well, it complained about _/root/.ansible/tmp/ansible-tmp-1472849250.29-49307224828763/source_, and not about _blah.tar.gz_. So maybe the file didn't transfer correctly, or it got corrupted somehow ? needs_info

**Comment by leonardolang** [comment]
Ok, so seems the mistake was on my side - I just found "tar" was not reachable via default PATH for the root user. I fixed it and everything seems to be working fine now. Thanks for your help!

**Comment by dagwieers** [comment]
@leonardolang Thanks for clarifying. Another ticket in the list we can close !!

PS Regarding your feedback related to the error-message, I am contemplating improving that message. Related to #3974.

**Comment by morgajel** [comment]
I just wanted to mention this (or something similar) bit me as well on a fresh install of Centos-7 via vagrant; it strikes me as weird that /bin/tar wouldn't be caught in root's path.

**Comment by dagwieers** [comment]
@morgajel I'd like to get more information regarding this as well.

Was PATH empty ? If so we could test if PATH is empty, **emit a warning** to the user and stuff it with some safe defaults. If not, there's little we can do I presume, but why is PATH not including _/bin_ ? Questions we need answers to...

**Comment by morgajel** [comment]
I've since worked around it by using command; what flags/data would be helpful?
